arch/rv32i/src/pmp.rs

```diff
@@ -128,20 +133,22 @@ pub struct PMPConfig {
     app_region: OptionalCell<usize>,
 }
 
-impl Default for PMPConfig {
+impl PMPConfigType for [Option<PMPRegion>; $x] {}
+
+impl Default for PMPConfig<[Option<PMPRegion>; $x]> {
     /// number of regions on the arty chip
-    fn default() -> PMPConfig {
+    fn default() -> Self {
         PMPConfig {
-            regions: [None; 32],
-            total_regions: 8,
+            regions: [None; $x],
+            total_regions: $x / 2,
```

Comment:
Could there be a comment indicating why this is `$x / 2`?

Reply:
I am adding comments.

Reply:
I didn't see this, I just pushed an update which doesn't remove the comments.